### PR TITLE
[quant][graphmode][fix] dequantize propagation for {add/mul}_scalar

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#40596 [quant][graphmode][fix] dequantize propagation for {add/mul}_scalar**

Summary:
Previously the fusion patterns for {add/mul}_scalar is inconsistent since the op pattern
produces a non-quantized tensor and the op replacement graph produces a quantized tensor

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22251072](https://our.internmc.facebook.com/intern/diff/D22251072)

### CODE REVIEW

File: test/quantization/test_quantize_jit.py [hunk 1]
```diff
@@ -2272,6 +2272,10 @@ def __init__(self):
 
             def forward(self, x):
                 x = self.conv(x)
+                # add_scalar
```
Comment (raghuramank100) [inline test/quantization/test_quantize_jit.py:2275]: I thought this was already supported. What was missing?

Comment (jerryzh168) [reply test/quantization/test_quantize_jit.py:2275]: the fusion pattern is inconsistent

File: torch/csrc/jit/passes/quantization/quantization_patterns.h [hunk 2]
```diff
@@ -504,29 +527,15 @@ graph(%a_quant, %b_quant, %alpha, %scale, %zero_point, %dtype):
...
-  std::string add_scalar = R"(
-graph(%a_quant, %b_scalar, %alpha):
-         %a_dequant = aten::dequantize(%a_quant)
-         %r = aten::add(%a_dequant, %b_scalar, %alpha)
-         return (%r) )";
-
-  std::string quantized_add_scalar = R"(
-graph(%a_quant, %b_scalar, %alpha):
-         %r = quantized::add_scalar(%a_quant, %b_scalar)
-         return (%r) )";
```
Comment (jerryzh168) [inline torch/csrc/jit/passes/quantization/quantization_patterns.h:517]: @raghuramank100 major change is this

Comment (jerryzh168) [reply torch/csrc/jit/passes/quantization/quantization_patterns.h:517]: original pattern produces a float tensor and quantized pattern produces a quantized tensor

**Review by raghuramank100 — APPROVED** [review]

File: torch/csrc/jit/passes/quantization/quantization_patterns.h (hunk 2 above)
Comment (raghuramank100) [reply torch/csrc/jit/passes/quantization/quantization_patterns.h:517]: So is it correct to say that the new pattern is: deq->add-> = quantize_add->deq?

Comment (jerryzh168) [reply torch/csrc/jit/passes/quantization/quantization_patterns.h:517]: new pattern is:
dq - aten::add -quant --> quantized::add_scalar

**Pull request closed without merging**